### PR TITLE
stability: git event watch and visualization

### PR DESCRIPTION
**Pull request by deads2k**
This helps solve problems with flakes in CI tests that can be due to operator problems and re-uses existing and valuable visualization for the run.

### CODE REVIEW

File: enhancements/kube-apiserver/stability-history-of-resources.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+When debugging a failed e2e test (or a string of them), one common question is, "what is the status of clusteroperator/foo
+when this particular test was running".
+While we could consider one-off solutions to this, we have a solution for storing this information inside of a local
```
Comment (stevekuznetsov) [inline enhancements/kube-apiserver/stability-history-of-resources.md:39]: Could you link to the tools metioned here? This document as written is so vague as to not be understandable :|

Comment (deads2k) [reply enhancements/kube-apiserver/stability-history-of-resources.md:39]: > Could you link to the tools metioned here? This document as written is so vague as to not be understandable :|

It captured the idea for @damemi who I think has found the tool and has a PR to add it.

Comment (damemi) [reply enhancements/kube-apiserver/stability-history-of-resources.md:39]: @stevekuznetsov the tool being referred to is https://github.com/mfojtik/ci-monitor-operator and we are working on adding it in https://github.com/openshift/origin/pull/24845

This has inspired a longer-term goal for me to add [distributed tracing[(https://github.com/openshift/enhancements/pull/279) throughout our components

File: enhancements/kube-apiserver/stability-history-of-resources.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+When combined with tooling that Seth Jennings built a year ago to visualize various rollout flows, we can have powerful
+correlative graphs.
+
+## Motivation
+
+David's clairvoyance isn't strong enough to see the past with clarity. 
+
+### Goals
+
+1. Know the state of clusteroperators, events, and pod at any given time.
```
Comment (lilic) [inline enhancements/kube-apiserver/stability-history-of-resources.md:52]: > Know the state of clusteroperators, events, and pod at any given time

What kind of state do you need to know, curious what metrics are missing that should be sent from CI clusters, as we plan on adding an ability to search through CI cluster metrics at some point in the near future.

File: enhancements/kube-apiserver/stability-history-of-resources.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+2. Visualize that state with Seth's tools via a simple link like promecieus.
+
+### Non-Goals
+
+1. Make a durable record intended for production, but that would be a really cool product.
+
+## Proposal
+
+1. Install Michal's tool in every cluster
```
Comment (lilic) [inline enhancements/kube-apiserver/stability-history-of-resources.md:61]: Currently we plan on enabling Prometheus remote write for CI clusters to send some metrics and alerts in pending state where they can be queried in a timeline. So should essentially do the same thing, right? https://docs.google.com/document/d/1_ILVUYNBC07EHaIlqel9EL1UCWLQlKlMJtTz2Xq9Tmo/edit

File: enhancements/kube-apiserver/stability-history-of-resources.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
+---
+title: stability-history-of-resources
+authors:
+  - "@deads2k"
+reviewers:
+approvers:
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
```
Comment (wking) [inline enhancements/kube-apiserver/stability-history-of-resources.md:9]: you need to pick values for these headers right?  Or should we just drop them from the template?

**Pull request closed without merging**